pdgraster/WMTSCapabilitiesGenerator.py

```diff
@@ -0,0 +1,247 @@
+        ET.SubElement(service_identification, "ows:Title").text = self.title
+        ET.SubElement(service_identification, "ows:ServiceType").text = "OGC WMTS"
+        ET.SubElement(service_identification, "ows:ServiceTypeVersion").text = "1.0.0"
+
+    def _add_operations_metadata(self, root):
+        operations_metadata = ET.SubElement(root, "ows:OperationsMetadata")
+        self._add_operation(operations_metadata, "GetCapabilities", self.capabilities_url)
+        self._add_operation(operations_metadata, "GetTile", self.tiles_url)
+
+    def _add_operation(self, parent, name, href):
+        operation = ET.SubElement(parent, "ows:Operation", attrib={"name": name})
+        dcp = ET.SubElement(operation, "ows:DCP")
+        http = ET.SubElement(dcp, "ows:HTTP")
+        get = ET.SubElement(http, "ows:Get", attrib={"xlink:href": href})
+        constraint = ET.SubElement(get, "ows:Constraint", attrib={"name": "GetEncoding"})
+        allowed_values = ET.SubElement(constraint, "ows:AllowedValues")
+        ET.SubElement(allowed_values, "ows:Value").text = "RESTful"
+
+    def _add_contents(self, root):
+        contents = ET.SubElement(root, "Contents")
+        layer = ET.SubElement(contents, "Layer")
+        ET.SubElement(layer, "ows:Title").text = "iwp_high"
+        ET.SubElement(layer, "ows:Identifier").text = "iwp_high"
+
+        wgs84_bbox = ET.SubElement(layer, "ows:WGS84BoundingBox")
+        ET.SubElement(wgs84_bbox, "ows:LowerCorner").text = f"{self.bounding_box['left']} {self.bounding_box['bottom']}"
+        ET.SubElement(wgs84_bbox, "ows:UpperCorner").text = f"{self.bounding_box['right']} {self.bounding_box['top']}"
+
+
+        style = ET.SubElement(layer, "Style", attrib={"isDefault": "true"})
+        ET.SubElement(style, "ows:Title").text = "Default Style"
+        ET.SubElement(style, "ows:Identifier").text = "default"
+
+        ET.SubElement(layer, "Format").text = "image/png"
+        tile_matrix_set_link = ET.SubElement(layer, "TileMatrixSetLink")
+        ET.SubElement(tile_matrix_set_link, "TileMatrixSet").text = "WGS1984Quad"
+
+        
+
+        resource_url = ET.SubElement(layer, "ResourceURL", attrib={
+            "format": self.tile_format,
+            "resourceType": "tile",
+            "template": self.resource_template
+        })
+
+        self._add_tile_matrix_set(contents)
+
+    def _add_tile_matrix_set(self, contents):
+        tile_matrix_set = ET.SubElement(contents, "TileMatrixSet", attrib={"xml:id": "WorldCRS84Quad"})
+        ET.SubElement(tile_matrix_set, "ows:Title").text = "CRS84 for the World"
+        ET.SubElement(tile_matrix_set, "ows:Identifier").text = "WGS1984Quad"
+
+        bounding_box = ET.SubElement(tile_matrix_set, "ows:BoundingBox", attrib={"crs": "http://www.opengis.net/def/crs/OGC/1.3/CRS84"})
+        ET.SubElement(bounding_box, "ows:LowerCorner").text = "-180 -90"
+        ET.SubElement(bounding_box, "ows:UpperCorner").text = "180 90"
```

Comment:
Are these corner coordinates always going to be global, even if the input bounding box is focused on a smaller area of interest?

Reply:
That's a good point. I changed the code slightly by making `bounding_box` an optional parameter. If it is passed, the area of interest will be used here - however, if left `None` default values will be applied.
```

 {   "left": -180,
    "right": 180,
    "bottom": -90,
    "top": 90
}
```